package/gluon-web-private-wifi/luasrc/lib/gluon/config-mode/model/admin/privatewifi.lua

```diff
@@ -13,9 +14,27 @@ local s = f:section(Section, nil, translate(
 	.. 'at the same time.'
 ))
 
+local uplink_interfaces = util.get_role_interfaces(uci, 'uplink')
+local mesh_on_wan = false
+
+for _, iface in ipairs(util.get_role_interfaces(uci, 'mesh')) do
+	if util.contains(uplink_interfaces, iface) then
+		mesh_on_wan = true
+	end
+end
+
 local enabled = s:option(Flag, "enabled", translate("Enabled"))
 enabled.default = uci:get('wireless', primary_iface) and not uci:get_bool('wireless', primary_iface, "disabled")
 
+if mesh_on_wan and enabled.data then
+	local w = Warning()
+	w:setcontent(translate(
+		'Meshing on WAN interface is enabled.' ..
+		'This can lead to problems.'
+	))
+	s:append(w)
```

Comment:
Form nodes must never be added conditionally, as it will break the ID numbering. Instead, always add the node and make the template show the div conditionally (for example only when a content has been set).

Reply:
Instead of adding more classes, the `Template` class can be used. Adding trivial setter methods is also unusual for `gluon-web-model`.

Something like the following should work:

```lua
let warning = s:option(Template, 'model/warning')
if mesh_on_wan and enabled.data then
	w.content = translate(
		'Meshing on WAN interface is enabled.' ..
		'This can lead to problems.'
	)
end
```

Reply:
I'm getting this error

```
Failed to execute dispatcher target for entry '/admin/privatewifi'.
The called action terminated with an exception:
/usr/lib/lua/gluon/web/model/classes.lua:423: class must be a descendant of AbstractValue
```

Reply:
Hmm, it seems that `Template` is currently meant to be used as a section rather than an option. So using
```lua
local warning = f:section(Template, 'model/warning')
```
before creating the section `s` might work, if that position of the warning looks okay.

If we really want it inside the existing section, allowing `Template` in addition to `AbstractValue` for `s:option()` would be acceptable.

Reply:
My other usecase involves having it next to the field that's causing the warning, so having it inside the section would be preferable.

Reply:
A custom method `s:template('model/warning', {content="..."})` perhaps?